Code/Framework/AzCore/AzCore/Debug/Trace.h

```diff
@@ -262,17 +261,18 @@ namespace AZ
     #define AZ_VerifyWarning(window, expression, ...) AZ_Warning(window, 0 != (expression), __VA_ARGS__)
 
 #else // !AZ_ENABLE_TRACING
-    #define AZ_Assert(expression, ...)
-    #define AZ_Error(window, expression, ...)
-    #define AZ_ErrorOnce(window, expression, ...)
-    #define AZ_Warning(window, expression, ...)
-    #define AZ_WarningOnce(window, expression, ...)
-    #define AZ_TracePrintf(window, ...)
-    #define AZ_TracePrintfOnce(window, ...)
-
-    #define AZ_Verify(expression, ...) (void)(expression)
-    #define AZ_VerifyError(window, expression, ...) (void)(expression)
-    #define AZ_VerifyWarning(window, expression, ...) (void)(expression)
+
+    #define AZ_Assert(...)               AZ_UNUSED(__VA_ARGS__);
+    #define AZ_Error(...)                AZ_UNUSED(__VA_ARGS__);
+    #define AZ_ErrorOnce(...)            AZ_UNUSED(__VA_ARGS__);
+    #define AZ_Warning(...)              AZ_UNUSED(__VA_ARGS__);
+    #define AZ_WarningOnce(...)          AZ_UNUSED(__VA_ARGS__);
+    #define AZ_TracePrintf(...)          AZ_UNUSED(__VA_ARGS__);
+    #define AZ_TracePrintfOnce(...)      AZ_UNUSED(__VA_ARGS__);
+
+    #define AZ_Verify(...)               AZ_UNUSED(__VA_ARGS__);
+    #define AZ_VerifyError(...)          AZ_UNUSED(__VA_ARGS__);
+    #define AZ_VerifyWarning(...)        AZ_UNUSED(__VA_ARGS__);
```

Comment:
The difference between `AZ_Verify*` and the regular assert/error/warning is that the expression in the `AZ_Verify*` functions is always executed, even in release. This change is effectively going to be break all uses of `AZ_Verify*` in release.

Reply:
The original code:
```
#define AZ_Verify(expression, ...) (void)(expression)
#define AZ_VerifyError(window, expression, ...) (void)(expression)
#define AZ_VerifyWarning(window, expression, ...) (void)(expression)
```
The exact same ends up happening in the new code, the call to `AZ_UNUSED` will expand that argument to `(void)(expression)`
However, in release, because of the optimization levels, unless `expression` modifies state, is likely that the compiler will optimize it out. That was also true for the previous code.